File: algorithms/java/Challenge_Coin_Change_2.java
class Solution {
    public int change(int amount, int[] coins) {
        int[] methods = new int[amount+1];
        methods[0] = 1;
        
        for(int coin: coins) {
            for(int i = coin ; i <= amount; ++i) {
                methods[i] += methods[i-coin];
            }
        }
        
        return methods[amount];
    }
}

File: algorithms/python/Challenge_Coin_Change_2.py
class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        methods = [0 for i in range(amount+1)]
        methods[0] = 1
        
        for coin in coins:
            for i in range(coin, amount+1):
                methods[i] += methods[i-coin]

        return methods[-1]
